applications/MeshingApplication/custom_utilities/tetrahedra10_mesh_converter_utility.h

```diff
@@ -0,0 +1,356 @@
+//
+
+#if !defined(KRATOS_TETRAHEDRA10_MESH_CONVERTER_UTILITY)
+#define  KRATOS_TETRAHEDRA10_MESH_CONVERTER_UTILITY
+
+// System includes
+
+/* Project includes */
+#include "custom_utilities/local_refine_tetrahedra_mesh.hpp"
+#include "geometries/tetrahedra_3d_10.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+///@}
+///@name Type Definitions
+///@{
+///@}
+///@name  Enum's
+///@{
+///@}
+///@name  Functions
+///@{
+///@}
+///@name Kratos Classes
+///@{
+class Tetrahedra10MeshConverter : public LocalRefineTetrahedraMesh
+{
+public:
+
+    ///@name Type Definitions
+    ///@{
+    ///@}
+
+    typedef Node<3> NodeType;
+    typedef Geometry<NodeType> GeometryType;
+    typedef GeometryType::Pointer GeometryPtrType;
+
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructors
+    Tetrahedra10MeshConverter(ModelPart& model_part) : LocalRefineTetrahedraMesh(model_part)
+    {
+
+    }
+
+    /// Destructor
+    ~Tetrahedra10MeshConverter() 
+    = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+        
+    /**
+    * Replaces the geometry of the mesh of Tetrahedra3D4 elements. Resulting is a mesh of Tetrahedra3D10 created
+    * by adding intermediate nodes to each Tetrahedra3D4, with same ids as the originals. Does the same for conditions,
+    * replacing the Triangle3D3 by Triangle3D6
+    * @param refine_on_reference: Boolean that defines if refine or not the mesh according to the reference
+    * @param interpolate_internal_variables: Boolean that defines if to interpolate or not the internal variables
+    */
+    void LocalConvertTetrahedra10Mesh(bool refine_on_reference, bool interpolate_internal_variables) {
+            for (auto element : mModelPart.Elements()) element.SetValue(SPLIT_ELEMENT,true);
+            for (auto condition : mModelPart.Conditions()) condition.SetValue(SPLIT_ELEMENT,true);
+            LocalRefineMesh(refine_on_reference, interpolate_internal_variables);
+        } 
+  
+    
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+    //int mPreviousRefinementLevel;
+    //int mCurrentRefinementLevel;
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+    ///@}
+
+private:
+    ///@name Private static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+    
+    /**
+    * Creates a new tetrahedra3D10
+    * @return the reference to the new tetrahedra
+    */
+    Tetrahedra3D10<Node<3>> GenerateTetrahedra(ModelPart& this_model_part, std::vector<int>& node_ids) {
```

Comment:
```suggestion
    Tetrahedra3D10<Node<3>> GenerateTetrahedra(
        ModelPart& rModelPart,
        std::vector<int>& rNodeIds) {
```
(minor) Multiline for more than one argument methods/functions and camel case arguments. Also note the `r` referring to reference in front.